### ISSUE TITLE
shift-click broken

### ISSUE THREAD

**Issue by billstclair** [tekezo/Karabiner-Elements#5]
I just installed the latest Karabiner-Elements on my iMac running the Sierra GM. I love that the "fn" key once again works as left control, but shift-click no longer works to extend the selection. Reasonable trade-off, but slightly annoying.

Here's my karabiner.json:

```
{
    "profiles": [
        {
            "name": "Default profile",
            "selected": true,
            "simple_modifications": {
                "fn": "left_control"
            }
        }
    ]
}
```

**Comment by billstclair** [comment]
It also breaks command-click and command-shift-click, which I use all the time in Firefox, to open a link in a new tab. That is more than annoying.

**Comment by oschrenk** [comment]
On my end Cmd Click is also not working for Chrome. Event Viewer doesn't give any inside.

**Comment by billstclair** [comment]
Strangely, Karabiner-EventViewer sees the Cmd and Shift flags on mouse clicks, when Karabiner-Elements is NOT running. Start Karabiner-Elements, however, and those flags no longer appear in Karabiner-EventViewer mouse clicks, though the FlagsChanged events still happen for the command and Shift keys.

**Comment by niieani** [comment]
Indeed, same problem here. Command+Click for opening new tabs stopped working for me too.

**Comment by mihkels** [comment]
Similar issue for me when Ctrl + Mouse swipe up (full screen zoom in). Is it possible that all key + mouse action combinations are broken?

**Comment by tokyovigilante** [comment]
Hi, the majority of modifier keys don't work for me after installing the nightly build and remapping right option -> forward delete on my 2015 MBP. Alt-Tab for task switching works, but Ctrl-Tab in Safari doesn't (just tabs between links on page instead of between tabs, nor does Ctrl + mouse zoom to zoom in screen.

**Comment by TSFoster** [comment]
Similar issue: option + click doesn't work. E.g. when clicking on the Airport icon in the menubar to view extra network information

**Comment by khallmark** [comment]
Same issue.

**Comment by leira** [comment]
Same issue here, mod keys with click doesn't work.

**Comment by tobsn** [comment]
Where does the karabiner.json go? I can't find anything where one would actually put it...

**Comment by talha131** [comment]
@tobsn Does [this link](https://github.com/tekezo/Karabiner-Elements/blob/master/usage/README.md#change-key) help?

**Comment by tobsn** [comment]
@talha131 yeah i found that a minute after asking when I decided to look into the code to find the part where it initially loads the json... *facepalm

thanks ;)

btw. combinations are not possible atm right? like command_r+7

**Comment by talha131** [comment]
> btw. combinations are not possible atm right? like command_r+7

Yeah they are not until #16 gets fixed.

**Comment by tekezo** [comment]
Thank you for feedback!
This issue is caused by OS X problem, and it needed large changes to avoid this problem.

I fixed this issue at Karabiner-Elements v0.90.22.
Please upgrade to the latest version from "Misc > Check for updates" or 
https://pqrs.org/latest/karabiner-elements-latest.dmg

**Comment by talha131** [comment]
I can confirm this issue is fixed.

**Comment by tokyovigilante** [comment]
Me too, thanks!

**Comment by billstclair** [comment]
I updated via "Misc > Check for updates". It took a long time, but finally brought up the app and asked me to identify my keyboard by pressing the key to the right of the left "shift" key. I did so, and nothing happened. I closed the dialog with the red circle on its upper-left-hand corner.

Karabiner-Elements then worked as expected, swapping the fn and control keys on my keyboard, as I specified in ~/.karabiner.d/configuration/karabiner.json. And shift and command work again with mouse clicks.

I can't use it, however, since it breaks things I rely on in Emacs. But I'll open a new issue for that, since it will effect only a small audience, and close this one.

**Comment by runar** [comment]
I had the same issue as @billstclair. My keyboard was not identified, but after closing the dialog everything worked as expected.

**Comment by talha131** [comment]
> but finally brought up the app and asked me to identify my keyboard by pressing the key to the right of the left "shift" key. I did so, and nothing happened. I closed the dialog with the red circle on its upper-left-hand corner.

I too faced the same issue. I closed the identify keyboard dialog, uninstalled old version of Karabiner-Elements, and did a fresh install.

**Comment by jphenow** [comment]
Can confirm, I had to do a manual re-install. Updated yesterday and still wasn't working, but a fresh reinstall today fixed the issue.

**Comment by niieani** [comment]
Looks like this is still not fully fixed. Randomly, after many hours of work, it sometimes stops working. Turning Karabiner Elements off and on again fixes it for some time, but it breaks again after a couple of hours.

**Comment by jphenow** [comment]
Was noticing the same thing today ☝️

**Comment by DenKoren** [comment]
I created new issue #204 for this problem, since discussing it in closed one not seems to be a good idea.
